tfjs-backend-webgpu/src/shader_preprocessor.ts

```diff
@@ -231,12 +231,12 @@ function getSetOutputSnippet(
       void setOutput(${dims.map(d => `int ${d}`).join(', ')}, vec4 value) {
         int flatIndex = getFlatIndex(${type}(${dims.join(', ')}), ${
           getShapeCoords(outShape)});
-        setOutput(flatIndex, value);
+        setOutput(flatIndex / 4, value);
       }
       void setOutput(${dims.map(d => `int ${d}`).join(', ')}, ivec4 value) {
         int flatIndex = getFlatIndex(${type}(${dims.join(', ')}), ${
           getShapeCoords(outShape)});
-        setOutput(flatIndex, value);
```

Comment:
Was this not used anywhere else? Or was the bug just not noticed until now?

Reply:
It was a bug just noticed. It could be triggered when you call it by coordinates `setOutput(coords[0], coords[1], coords[2], value)` . Currently, it hasn't been used anywhere. Since the change is very simple, I put it here.